apps/ui/src/views/SpaceUser/Votes.vue

```diff
@@ -1,5 +1,144 @@
-<script setup lang="ts"></script>
+<script setup lang="ts">
+import { getNetwork } from '@/networks';
+import { Proposal, Space, User, Vote } from '@/types';
+
+const props = defineProps<{ space: Space; user: User }>();
+
+const PAGINATION_LIMIT = 20;
+const VOTES_LIMIT = 1000;
+
+const metaStore = useMetaStore();
+const { setTitle } = useTitle();
+
+const loaded = ref(false);
+const loadingMore = ref(false);
+const hasMore = ref(false);
+const failed = ref(false);
+const proposals = ref<Proposal[]>([]);
+const votes = ref<Record<Proposal['id'], Vote>>({});
+
+const network = computed(() => getNetwork(props.space.network));
+
+async function loadVotes() {
+  try {
+    votes.value = await network.value.api.loadUserVotes(
+      [props.space.id],
+      props.user.id,
+      {
+        limit: VOTES_LIMIT
+      }
+    );
+  } catch (e) {
+    failed.value = true;
+  }
+}
+
+function getProposalIds(offset = 0): string[] {
+  return Object.keys(votes.value)
+    .slice(offset, offset + PAGINATION_LIMIT)
+    .map(id => id.split(':')[1]);
+}
+
+async function fetch() {
+  loaded.value = false;
+  const ids = getProposalIds();
+
+  if (ids.length) {
+    proposals.value = await network.value.api.loadProposals(
+      [props.space.id],
+      {
+        limit: PAGINATION_LIMIT
+      },
+      metaStore.getCurrent(props.space.network) || 0,
+      { id_in: ids }
+    );
+
+    hasMore.value = proposals.value.length === PAGINATION_LIMIT;
+  }
+
+  loaded.value = true;
+}
+
+async function fetchMore() {
+  const ids = getProposalIds(proposals.value.length);
+
+  if (!ids.length) {
+    hasMore.value = false;
+    return;
+  }
+
+  loadingMore.value = true;
+  const moreProposals = await network.value.api.loadProposals(
+    [props.space.id],
+    {
+      limit: PAGINATION_LIMIT
+    },
+    metaStore.getCurrent(props.space.network) || 0,
+    { id_in: ids }
+  );
+
+  proposals.value = [...proposals.value, ...moreProposals];
+  hasMore.value = moreProposals.length === PAGINATION_LIMIT;
+  loadingMore.value = false;
+}
+
+async function handleEndReached() {
+  if (!hasMore.value) return;
+
+  fetchMore();
+}
+
+onMounted(async () => {
+  await loadVotes();
+  fetch();
+});
```

Comment:
`onMounted` usually is problematic in cases like this, because if we navigate to the same page, but for different space (I don't think it's possible via navigation, but you can change URL and it won't trigger) we won't call it again.

We would need to watch space/user instead.

Reply:
I don't think it applies to this case, since the Space and User are passed to this view as props.

The change in `userId` is already watched by the parent SpaceUser, and any changes in `spaceId` is watched by the grandparent Space

Changing the url, and navigating between different user and space are refreshing the content as expected